src/libs/banners/banners.ts

```diff
@@ -69,59 +69,92 @@ export const getBridgeBanners = (
     })
   }
 
-  return activeRoutes
-    .filter(isBridgeTxn)
-    .filter((route) => {
-      if (route.routeStatus === 'failed') return false
-      if (route.routeStatus !== 'ready') return true
-      // If the route is ready to be signed, we should display the banner only if it's not turned into an account op
-      // because when it does get turned into an account op, there will be a different banner for that
-      return !isRouteTurnedIntoAccountOp(route)
-    })
-    .map((r) => {
-      const actions: Action[] = []
+  const filteredRoutes = activeRoutes.filter(isBridgeTxn).filter((route) => {
+    if (route.routeStatus === 'failed') return false
+    if (route.routeStatus !== 'ready') return true
+    return !isRouteTurnedIntoAccountOp(route)
+  })
+
+  const inProgressRoutes = filteredRoutes.filter(
+    (r) => r.routeStatus === 'in-progress' || r.routeStatus === 'waiting-approval-to-resolve'
+  )
 
-      if (r.routeStatus === 'in-progress' || r.routeStatus === 'waiting-approval-to-resolve') {
-        actions.push({
+  const completedRoutes = filteredRoutes.filter((r) => r.routeStatus === 'completed')
+
+  const remainingRoutes = filteredRoutes.filter(
+    (r) => r.routeStatus !== 'in-progress' && r.routeStatus !== 'completed'
+  )
```

Comment:
`failed` routes are filtered immediately, so we are left with `'waiting-approval-to-resolve' | 'in-progress' | 'ready' | 'completed'`. After filtering out `in-progress` and `completed` the remaining routes are `waiting-approval-to-resolve` and `ready` but `waiting-approval-to-resolve` is included in `inProgressRoutes`. Wouldn't there be duplicate banners?

Reply:
Correct, thank you!